### PR TITLE
doc: redirects: regroup and add comments

### PR DESCRIPTION
**Pull request by greg-fer**
Regrouped redirect tuples based on page hierarchy. Added comments with page names.
Updated notes in the description at the top of the file.

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

File: doc/_utils/redirects.py [hunk 1]
```diff
@@ -4,455 +4,474 @@
...
+    ("samples/samples_matter","samples/matter"), # Matter samples (landing)
+    ("samples/samples_nfc","samples/nfc"), # NFC samples (landing)
+    ("samples/samples_nrf5340","samples/nrf5340"), # nRF5340 samples (landing)
+    ("samples/samples_thread","samples/thread"), # Thread samples (landing)
+    ("samples/samples_tfm","samples/tfm"), # Trusted Firmware-M (TF-M) samples (landing)
+    ("samples/samples_wifi","samples/wifi"), # Wi-Fi samples (landing)
+    ("samples/wifi/sr_coex/README","samples/wifi/ble_coex/README"), # Wi-Fi: Bluetooth LE coexistence
+    ("samples/samples_zigbee","samples/zigbee"), # Zigbee samples (landing)
+    ("samples/samples_other","samples/other"), # Other samples (landing)
+    ("libraries/bin/bt_ll_acs_nrf53/index", "nrfxlib/softdevice_controller/doc/isochronous_channels"), # LE Audio controller for nRF5340 (removed for 2.7.0)
```
Comment (umapraseeda) [inline doc/_utils/redirects.py:421]: This redirection is not working, it resolves to https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-16834/nrf/nrfxlib/softdevice_controller/doc/isochronous_channels.html

Comment (greg-fer) [reply doc/_utils/redirects.py:421]: @gmarull , is it OK to use `../` in these tuples?

File: doc/_utils/redirects.py [hunk 2]
```diff
@@ -4,455 +4,474 @@
...
+    ("ug_matter_gs_testing","protocols/matter/getting_started/testing/index"), # Testing Matter in the nRF Connect SDK (landing)
+    ("ug_matter_gs_testing_thread_separate_linux_macos","protocols/matter/getting_started/testing/thread_separate_otbr_linux_macos"), # Matter over Thread: Configuring Border Router and Linux/macOS controller on separate devices
+    ("ug_matter_gs_testing_thread_one_otbr","protocols/matter/getting_started/testing/thread_one_otbr"), # Matter over Thread: Configuring Border Router and controller on one device
+    ("ug_matter_gs_testing_wifi_pc","protocols/matter/getting_started/testing/wifi_pc"), # Matter over Wi-Fi: Configuring CHIP Tool for Linux or macOS
+    ("ug_matter_gs_testing_thread_separate_otbr_android","protocols/matter/getting_started/testing/thread_separate_otbr_android"), # Matter over Thread: Configuring Border Router and Android controller on separate devices (removed after 2.2.0)
+    ("ug_matter_gs_testing_wifi_mobile","protocols/matter/getting_started/testing/wifi_mobile"), # Matter over Wi-Fi: Configuring CHIP Tool for Android (removed after 2.2.0)
```
Comment (umapraseeda) [inline doc/_utils/redirects.py:353]: Since it is removed now, is the redirection useful?

File: doc/_utils/redirects.py [hunk 3]
```diff
@@ -4,455 +4,474 @@
...
+    ("ug_gzll","protocols/gazell/gzll"), # Gazell Link Layer
+    ("ug_gzp","protocols/gazell/gzp"), # Gazell Pairing
+    ("ug_matter","protocols/matter/index"), # Matter (landing)
+    ("ug_matter_intro_overview","protocols/matter/overview/index"), # Matter overview (landing)
+    ("ug_matter_overview_architecture","protocols/matter/overview/architecture"), # Matter architecture
+    ("ug_matter_overview_data_model","protocols/matter/overview/data_model"), # Matter Data Model and device types
+    ("ug_matter_overview_int_model","protocols/matter/overview/int_model"), # Matter Interaction Model and interaction types
+    ("ug_matter_overview_network_topologies","protocols/matter/overview/network_topologies"), # Matter network topology and concepts
+    ("ug_matter_overview_security","protocols/matter/overview/security"), # Matter network security
+    ("ug_matter_overview_commissioning","protocols/matter/overview/commisioning"), # Matter network commissioning
```
Comment (umapraseeda) [inline doc/_utils/redirects.py:341]:
```suggestion
    ("ug_matter_overview_commissioning","protocols/matter/overview/commissioning"), # Matter network commissioning
```

File: doc/_utils/redirects.py [hunk 4]
```diff
@@ -4,455 +4,474 @@
...
+    ("protocols/bt_mesh/configuring", "protocols/bt/bt_mesh/configuring"),
+    ("ug_bt_mesh_model_config_app", "protocols/bt_mesh/model_config_app"), # Configuring mesh models using the nRF Mesh mobile app
+    ("protocols/bt_mesh/model_config_app", "protocols/bt/bt_mesh/model_config_app"),
+    ("ug_bt_mesh_fota", "protocols/bt_mesh/fota"), # Performing Device Firmware Updates (DFU) in Bluetooth Mesh
+    ("protocols/bt_mesh/fota", "protocols/bt/bt_mesh/fota"),
+    ("ug_bt_mesh_node_removal", "protocols/bt_mesh/node_removal"), # Removing a node from a mesh network
+    ("protocols/bt_mesh/node_removal", "protocols/bt/bt_mesh/node_removal"),
+    ("ug_bt_mesh_vendor_model", "protocols/bt_mesh/vendor_model/index"), # Creating a new model (landing)
+    ("protocols/bt_mesh/vendor_model/index", "protocols/bt/bt_mesh/vendor_model/index"),
+    ("ug_bt_mesh_vendor_model_dev_overview", "protocols/bt_mesh/vendor_model/dev_overview"), # Vendor model development overview
```
Comment (umapraseeda) [inline doc/_utils/redirects.py:324]:
```suggestion
    ("ug_bt_mesh_vendor_model_dev_overview", "protocols/bt/bt_mesh/vendor_model/dev_overview"), # Vendor model development overview
```

File: doc/_utils/redirects.py (hunk 3 above)
Comment (greg-fer) [reply doc/_utils/redirects.py:341]: Hm, I swear I fixed that...

File: doc/_utils/redirects.py (hunk 4 above)
Comment (greg-fer) [reply doc/_utils/redirects.py:324]: This is incorrect. The edit is needed in the next line, and I have now applied it.

Comment (umapraseeda) [reply doc/_utils/redirects.py:324]: I see. Yes, thought it was this line as the redirection failed.

File: doc/_utils/redirects.py [hunk 5]
```diff
@@ -4,455 +4,474 @@
...
+    ("device_guides/nrf53/fota_update_nrf5340", "app_dev/device_guides/nrf53/fota_update_nrf5340"), # FOTA updates with nRF5340 DK
+    ("device_guides/nrf53/multi_image_nrf5340", "app_dev/device_guides/nrf53/multi_image_nrf5340"), # Multi-image builds on the nRF5340 DK using child and parent images
+    ("device_guides/nrf53/simultaneous_multi_image_dfu_nrf5340", "app_dev/device_guides/nrf53/simultaneous_multi_image_dfu_nrf5340"), # Simultaneous multi-image DFU with nRF5340 DK
+    ("device_guides/nrf53/serial_recovery", "app_dev/device_guides/nrf53/serial_recovery"), # MCUboot’s serial recovery of the networking core image
+    ("device_guides/nrf53/logging_nrf5340", "app_dev/device_guides/nrf53/logging_nrf5340"), # Getting logging output with nRF5340 DK
+    ("device_guides/nrf53/thingy53_application_guide", "app_dev/device_guides/nrf53/thingy53_application_guide"), # Application guide for Thingy:53
+    ("device_guides/working_with_nrf/nrf53/qspi_xip_guid", "device_guides/nrf53/qspi_xip_guide_nrf5340"), # External execute in place (XIP) configuration on the nRF5340 SoC
```
Comment (umapraseeda) [inline doc/_utils/redirects.py:248]:
```suggestion
    ("device_guides/working_with_nrf/nrf53/qspi_xip_guide", "device_guides/nrf53/qspi_xip_guide_nrf5340"), # External execute in place (XIP) configuration on the nRF5340 SoC
```

File: doc/_utils/redirects.py [hunk 6]
```diff
@@ -4,455 +4,474 @@
...
+    ("config_and_build/dfu/index", "app_dev/dfu/index"), # Device Firmware Updates (removed after 2.7.0) > Bootloaders and DFU (landing)
+    ("app_dev/dfu/index", "app_dev/bootloaders_dfu/index"),
+    ("app_bootloaders", "config_and_build/bootloaders_and_dfu/index"), # Bootloaders (removed after 2.7.0) > Bootloaders and DFU (landing)
+    ("app_dev/bootloaders_and_dfu/index", "config_and_build/bootloaders_and_dfu/index"),
+    ("config_and_build/bootloaders_and_dfu/index", "config_and_build/bootloaders_dfu/index"),
+    ("config_and_build/bootloaders_dfu/index", "app_dev/bootloaders_dfu/index"),
+    ("config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_mcuboot_nsib", "app_dev/bootloaders_dfu/mcuboot_nsib/bootloader_mcuboot_nsib"), # MCUboot and NSIB (landing)
+    ("ug_bootloader_testing", "config_and_build/bootloaders_and_dfu/bootloader_testing"), # Testing the bootloader chain (removed after 2.7.0)
+    ("app_dev/bootloaders_and_dfu/bootloader_testing", "config_and_build/bootloaders_and_dfu/bootloader_testing"),
+    ("app_dev/bootloaders_and_dfu/bootloader_testing", "config_and_build/bootloaders/bootloader_testing"),
```
Comment (umapraseeda) [inline doc/_utils/redirects.py:106]: 2 entries for the same source. Not working. Is it supposed to be
```suggestion
    ("app_dev/bootloaders_and_dfu/bootloader_testing", "config_and_build/bootloaders_and_dfu/bootloader_testing"),
    ("config_and_build/bootloaders_and_dfu/bootloader_testing", "config_and_build/bootloaders/bootloader_testing"),
```

File: doc/_utils/redirects.py [hunk 7]
```diff
@@ -4,455 +4,474 @@
...
+    ("config_and_build/bootloaders/bootloader_testing", "config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_mcuboot_nsib"),
+    ("config_and_build/bootloaders/bootloader_quick_start", "config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_quick_start"), # Quick start guide
+    ("config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_quick_start", "app_dev/bootloaders_dfu/mcuboot_nsib/bootloader_quick_start"),
+    ("config_and_build/bootloaders/bootloader_adding_sysbuild", "config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_adding_sysbuild"), # Enabling a bootloader chain using sysbuild
+    ("config_and_build/bootloaders_dfu/mcuboot_nsib/bootloader_adding_sysbuild", "app_dev/bootloaders_dfu/mcuboot_nsib/bootloader_adding_sysbuild"),
+    ("ug_bootloader_adding", "config_and_build/bootloaders_and_dfu/bootloader_adding"), # "Enabling a bootloader chain using child and parent images (deprecated)"
     ("app_dev/bootloaders_and_dfu/bootloader_adding", "config_and_build/bootloaders_and_dfu/bootloader_adding"),
     ("app_dev/bootloaders_and_dfu/bootloader_adding", "config_and_build/bootloaders/bootloader_adding"),
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/_utils/redirects.py:114]: No, the issue is lack of app_dev (2nd entry) in first line and config_and_build in third line (1st entry).

File: doc/_utils/redirects.py (hunk 1 above)
Comment (umapraseeda) [reply doc/_utils/redirects.py]: Seems to be working.

**Review by umapraseeda — APPROVED** [review]
Good cleanup. Checked all the redirects. Thanks!

**Pull request merged**